packages/ns-api/files/ns.ovpnrw

```diff
@@ -882,6 +886,120 @@ def download_user_2fa(ovpninstance, username):
     else:
         return utils.validation_error("username", "2fa_download_failed", username)
 
+def _get_database(ovpninstance):
+    # We try the ram path, in the future we could have a storage path, if fail we return an empty string
+    db_path = [ f'/var/openvpn/{ovpninstance}/connections.db']
+    database_path = ""
+    for db in db_path:
+        if os.path.isfile(db):
+            database_path = db
+            break
+    return database_path
+
+def connection_history_csv(ovpninstance):
+    database_path = _get_database(ovpninstance)
+    if database_path == "":
+        return {}
+
+    conn = sqlite3.connect(database_path)
+    c = conn.cursor()
+
+    # Define the CSV file path
+    csv_directory = '/var/run/ns-api-server/downloads'
+    csv_file_path = os.path.join(csv_directory, 'connection_history.csv')
+
+    # Ensure the directory exists
+    os.makedirs(csv_directory, exist_ok=True)
+
+    with open(csv_file_path, 'w', newline='') as csv_file:
+        csv_writer = csv.writer(csv_file)
+        # Write the CSV header
+        csv_writer.writerow(["Started", "Ended", "Duration (seconds)", "User", "Virtual IP", "Remote IP", "Received (KB)", "Sent (KB)"])
+
+        try:
+            rows = c.execute('''SELECT virtual_ip_addr, remote_ip_addr, start_time, duration, bytes_received, bytes_sent, common_name
+                                FROM connections
+                                ORDER BY start_time DESC''')
+        except:
+            conn.close()
+            sys.exit(0)
+
+        for row in rows:
+            virtual_ip_addr = row[0]
+            remote_ip_addr = row[1]
+            start_time = row[2]
+            duration = row[3]
+
+            if row[4]:
+                bytes_received = format((float(row[4]) / 1024), '.2f')
+            else:
+                bytes_received = 0
+
+            if row[5]:
+                bytes_sent = format((float(row[5]) / 1024), '.2f')
+            else:
+                bytes_sent = 0
```

Comment:
Let's return the raw data: this is an export and it's aimed to do elaboration on a different software